presto-hive/src/main/java/io/prestosql/plugin/hive/HiveType.java

```diff
@@ -148,26 +152,32 @@ public String toString()
         return hiveTypeName.toString();
     }
 
-    public boolean isSupportedType()
+    public boolean isSupportedType(StorageFormat storageFormat)
     {
-        return isSupportedType(getTypeInfo());
+        return isSupportedType(getTypeInfo(), storageFormat);
     }
 
-    public static boolean isSupportedType(TypeInfo typeInfo)
+    public static boolean isSupportedType(TypeInfo typeInfo, StorageFormat storageFormat)
     {
         switch (typeInfo.getCategory()) {
             case PRIMITIVE:
                 return getPrimitiveType((PrimitiveTypeInfo) typeInfo) != null;
             case MAP:
                 MapTypeInfo mapTypeInfo = (MapTypeInfo) typeInfo;
-                return isSupportedType(mapTypeInfo.getMapKeyTypeInfo()) && isSupportedType(mapTypeInfo.getMapValueTypeInfo());
+                return isSupportedType(mapTypeInfo.getMapKeyTypeInfo(), storageFormat) && isSupportedType(mapTypeInfo.getMapValueTypeInfo(), storageFormat);
             case LIST:
                 ListTypeInfo listTypeInfo = (ListTypeInfo) typeInfo;
-                return isSupportedType(listTypeInfo.getListElementTypeInfo());
+                return isSupportedType(listTypeInfo.getListElementTypeInfo(), storageFormat);
             case STRUCT:
                 StructTypeInfo structTypeInfo = (StructTypeInfo) typeInfo;
                 return structTypeInfo.getAllStructFieldTypeInfos().stream()
-                        .allMatch(HiveType::isSupportedType);
+                        .allMatch(fieldTypeInfo -> isSupportedType(fieldTypeInfo, storageFormat));
+            case UNION:
+                if (storageFormat.getSerDe().equalsIgnoreCase(AVRO.getSerDe()) || storageFormat.getSerDe().equals(ORC.getSerDe())) {
```

Comment:
Add a comment explaining whyh this is storage format dependent

Reply:
Thanks, I've added such a comment